pkg/router/template/plugin.go

```diff
@@ -156,6 +156,10 @@ func (p *TemplatePlugin) HandleEndpoints(eventType watch.EventType, endpoints *k
 		if commit {
 			p.Router.Commit()
 		}
+	case watch.Deleted:
+		glog.V(4).Infof("Deleting endpoints for %s", key)
+		p.Router.DeleteEndpoints(key)
+		p.Router.Commit()
```

Comment:
Aah, ok - just saw the bugz re: service deletion as opposed to route deletion.  
Could you please add a test - the changes look good.  Thanks.